src/server/scripts/Outland/BlackTemple/boss_illidan.cpp

```diff
@@ -1298,12 +1298,21 @@ struct npc_flame_of_azzinoth : public ScriptedAI
             DoCastVictim(SPELL_FLAME_BLAST);
 
             me->m_Events.AddEventAtOffset([&] {
-                if (Unit* target = me->GetVictim())
-                    target->CastSpell(target, SPELL_BLAZE, true);
+                DoCastVictim(SPELL_BLAZE);
             }, 1s);
         }, 15s, 20s);
     }
 
+    void UpdateAI(uint32 diff) override
```

Comment:
overriding this is not needed, parent class will already do all of this

Reply:
That's what I thought, but without it, no cast for Blaze and Blame flame. Also no automatic DoMeleeAttackIfReady()